Doc/library/asyncio-stream.rst

```diff
@@ -206,12 +206,21 @@ StreamReader
 
    .. coroutinemethod:: read(n=-1)
 
-      Read up to *n* bytes.  If *n* is not provided, or set to ``-1``,
-      read until EOF and return all read bytes.
+      Read up to *n* bytes from the stream.
 
+      If *n* is not provided, or set to ``-1``,
+      read until EOF, then return all read :class:`bytes`.
       If EOF was received and the internal buffer is empty,
       return an empty ``bytes`` object.
 
+      If *n* is ``0``, return an empty ``bytes`` object immediately.
+
+      If *n* is positive, wait for at least 1 byte to become available
+      (or EOF), then return the available ``bytes`` which might be less
+      than *n*.
```

Comment:
if there's still data in the buffer, this doesn't wait. Maybe the text should allude to that?

Reply:
Yeah, I've gone ahead and included that in my suggestion